api/bazel/api_build_system.bzl

```diff
@@ -180,3 +181,60 @@ def api_go_test(name, size, importpath, srcs = [], deps = []):
         importpath = importpath,
         deps = deps,
     )
+
+_GO_BAZEL_RULE_MAPPING = {
+    "@opencensus_proto//opencensus/proto/trace/v1:trace_proto": "@opencensus_proto//opencensus/proto/trace/v1:trace_proto_go",
+    "@opencensus_proto//opencensus/proto/trace/v1:trace_config_proto": "@opencensus_proto//opencensus/proto/trace/v1:trace_and_config_proto_go",
+    "@com_google_googleapis//google/api/expr/v1alpha1:syntax_proto": "@com_google_googleapis//google/api/expr/v1alpha1:cel_go_proto",
+}
+
+def go_proto_mapping(dep):
+    mapped = _GO_BAZEL_RULE_MAPPING.get(dep)
+    if mapped == None:
+        return _Suffix("@" + Label(dep).workspace_name + "//" + Label(dep).package + ":" + Label(dep).name, _GO_PROTO_SUFFIX)
+    return mapped
+
+def api_proto_package(name, srcs = [], deps = [], has_services = False, visibility = ["//visibility:public"]):
+    native.proto_library(
+        name = name,
+        srcs = srcs,
+        deps = deps + [
+            "@com_google_protobuf//:any_proto",
+            "@com_google_protobuf//:descriptor_proto",
+            "@com_google_protobuf//:duration_proto",
+            "@com_google_protobuf//:empty_proto",
+            "@com_google_protobuf//:struct_proto",
+            "@com_google_protobuf//:timestamp_proto",
+            "@com_google_protobuf//:wrappers_proto",
+            "@com_google_googleapis//google/api:http_proto",
+            "@com_google_googleapis//google/api:annotations_proto",
+            "@com_google_googleapis//google/rpc:status_proto",
+            "@com_github_gogo_protobuf//:gogo_proto",
+            "@com_envoyproxy_protoc_gen_validate//validate:validate_proto",
+        ],
+        visibility = visibility,
+    )
+
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "//bazel:pgv_plugin_go"]
+    if has_services:
+        compilers = ["@io_bazel_rules_go//proto:go_grpc", "//bazel:pgv_plugin_go"]
+
+    go_proto_library(
```

Comment:
Can we delete `api_go_grpc_library` etc. now? This will prevent these reappearing later in BUILD files.

Reply:
Sure, once UDPA stuff is gone. I think you have a PR to delete it.

Reply:
Yep, https://github.com/envoyproxy/envoy/pull/8017

Reply:
Updated and purged old api_go rules.